src/shapes/ShapeAttributes.js

```diff
@@ -300,6 +300,25 @@ define([
             }
         });
 
+        ShapeAttributes.prototype.clone = function () {
```

Comment:
The approach at other places is to use `new ShapeAttributes(baseShapeAttribute)` not a clone method. However, I see that Color has a clone method. We aren't consistent everywhere. Moreover, we might want to clone the color in the constructor of the attribute classes.